docs/en/sql-reference/dictionaries/external-dictionaries/external-dicts-dict-layout.md

```diff
@@ -156,6 +156,33 @@ or
 LAYOUT(HASHED(PREALLOCATE 0))
 ```
 
+If `shards` greater then 1 (default is `1`) the dictionary will load data in parallel, useful if you have huge amount of elements in one dictionary.
+
+Configuration example:
+
+``` xml
+<layout>
+  <hashed>
+    <shards>10</shards>
+    <!-- Size of the backlog for blocks in parallel queue.
+
+         Since the bottleneck in parallel loading is rehash, and so to avoid
+         stalling because of thread is doing rehash, you need to have some
+         backlog.
+
+         10000 is good balance between memory and speed.
+         Even for 10e10 elements and can handle all the load without starvation. -->
+    <parallel_queue_backlog>10000</parallel_queue_backlog>
```

Comment:
This setting name is extemely hard to understand. Better to remove it completely or rename.

Reply:
Will rename it to `per_shard_load_backlog`